lib/ansible/modules/network/meraki/meraki_rest.py

```diff
@@ -0,0 +1,238 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2018, Kevin Breit (@kbreit) <kevin.breit@kevinbreit.net>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: meraki_rest
+
+short_description: Direct access to the Cisco Meraki management dashboard
+
+version_added: "2.6"
+
+description:
+    - Enables management of Cisco's Meraki line of products through their cloud based dashboard.
+    - More information about the Meraki API can be found at U(https://dashboard.meraki.com/api_docs).
+
+options:
+    auth_key:
+        description:
+            - Authentication key provided by the dashboard. Required if environmental variable MERAKI_KEY is not set.
+        required: no
+    host:
+        description:
+            - FQDN or IP address of Meraki dashboard.
+        required: yes
+    method:
+        description:
+            - The HTTP method of the request.
+            - Using C(delete) is typically used for deleting objects.
+            - Using C(get) is typically used for querying objects.
+            - Using C(put) is typically used for modifying objects.
+            - Using C(post) is typically used for modifying objects.
+        required: yes
+        choices: [ delete, get, post, put ]
+    path:
+        description:
+            - Directory path to the endpoint. Do not include FQDN specified in C(host).
+        required: yes
+    timeout:
+        description:
+            - HTTP timeout value.
+        default: 30
+    use_proxy:
+        description:
+            - If C(no), it will not use a proxy, even if one is defined in an environment variable on the target hosts.
+        type: bool
+        default: no
+    use_ssl:
+        description:
+            - If C(no), it will use HTTP. Otherwise it will use HTTPS.
+        type: bool
+        default: yes
+    validate_certs:
+        description:
+            - If C(no), HTTPS certificates will not be verified.
+        type: bool
+        default: yes
+    content:
+        description:
+            - Raw content which should be fed in body.
+    src:
+        description:
+            - Name of absolute path of the filename which contains content to be fed to Meraki Dashboard. Must be in JSON format.
+    output_level:
+        description:
+            - Set amount of debug output during module execution
+        choices: ['normal', 'debug']
+
+author:
+    - Kevin Breit (@kbreit)
+'''
+
+EXAMPLES = '''
+# Query inventory
+- name: Query network device inventory
+  sda_rest:
+    auth_key: abc12345
+    host: dashboard.meraki.com
```

Comment:
I would default to api.meraki.com, so that people don't have to provide this information.